### ISSUE TITLE
Different SINR(dB) ranges for devices in a same container

### ISSUE THREAD

**Issue by arturandre** [nyuwireless-unipd/ns3-mmwave#106]
Hello everyone,

I'm experimenting a scenario with 1 eNB and 40 devices. I tried to run the simulation for 20 seconds and I noticed that the SINR(dB) given a device, gets stuck in a certain range. So some devices always have a SINR(dB) above some value while others always have SINR(dB) below some value. I was expecting that since all the devices are in the same container (and position) then the SINR(dB) would be rather homogeneous for all the devices. Is this a normal thing or there's something wrong with my script?

This first image shows the SINR for all the devices across time. Notice that none of the lines vary from 0 to 50, all of them oscillate nearly 10 dB above or below its initial point. 

![SINR 40 devices](https://user-images.githubusercontent.com/1486993/57472470-e0c75780-7263-11e9-9e85-95840b451aaf.png)

This second image shows the SINR for a specific device, notice that the SINR ranges from 12 to 22 approximately, for the whole simulation.

![SINR device 1](https://user-images.githubusercontent.com/1486993/57472684-74992380-7264-11e9-8a0d-c44201e1e0da.png)

This third image shows another device, at the same container and position as the first one, and yet its SINR ranges from 30 to 40, that is, its lowest SINR is greater than the maximum SINR for another device in the same location.

![SINR device 5](https://user-images.githubusercontent.com/1486993/57473086-6992c300-7265-11e9-9964-5d7136926beb.png)

Here is the script (in .txt since github doesn't allow .cc).

[mmwave-example.txt](https://github.com/nyuwireless-unipd/ns3-mmwave/files/3163268/mmwave-example.txt)

Thanks in advance.

**Comment by MengleiZhang** [comment]
Can you try the new 3GPP channel, which can be enabled by adding the
following scripts after declaring *ptr_mmWave*.

ptr_mmWave ->SetAttribute ("PathlossModel", StringValue ("
ns3::MmWave3gppBuildingsPropagationLossModel"));
ptr_mmWave ->SetAttribute ("ChannelModel", StringValue ("
ns3::MmWave3gppChannel"));
Moreover, if you want the same SINR for all UE devices, try to disable the
shadowing in the propagation channel by adding the following,

Config::SetDefault ("ns3::MmWave3gppPropagationLossModel::Shadowing",
BooleanValue(false)); // enable or disable the shadowing effect
Menglei

**Comment by arturandre** [comment]
Thank you very Menglei.

When I tried to use the new 3GPP channel by inserting the code you provided just after declaring ptr_mmwave and just before initializing (after initializing I got another kind of error) it I got the error at the bottom of the message. I'm testing now the solution to get the same SINR for all UE, as soon as I get the results I'll report them here.

`bandwidth 0 = 1.00001e+09                                                                                            SetBeamformingVector between UE 0x56336405f180 and enbDevice 0x5633640f0bb0                                          RMa scenario, channel condition is fixed to be l, h_BS=0,h_UT=0                                                      msg="According to table 7.4.1-1, the RMa scenario need to satisfy the following condition, 10 m <= hBS <= 150 m", file=../src/mmwave/model/mmwave-3gpp-propagation-loss-model.cc, line=334                                                terminate called without an active exception                                                                         Command ['/ns3-mmwave/build/src/mmwave/examples/ns3-dev-mmwave-example-debug'] terminated with signal SIGIOT. Run it under a debugger to get more information (./waf --run <program> --command-template="gdb --args %s <args>").`

**Comment by arturandre** [comment]
Hi again. I got the results from turning the shadowing off and they still confusing.

This time I tried 40 devices connected to a single eNB and once again I got widely different SINR(dB). But the most strange part is that for some devices the SINR(dB) is greater than 42 dB for the whole simulation (5 seconds) while some others can't get over 15 dB for the whole simulation. I expected that since all the UEs are located in the same spot, and connected with the same eNB they would have at least the same variance in their SINR measures. I'll post a new plot with show the SINR for this test.

![image](https://user-images.githubusercontent.com/1486993/58370737-4eb88380-7ee0-11e9-8029-a87806ef682f.png)


One thing I believe is worth noting is that I got 88 RNTIs instead of just 40, that is, some UEs perhaps are associated with more than one single RNTI, so I suppose that a single UE "loses" connection and then when it get it back it is associated with a new RNTI, is that correct?

**Comment by MengleiZhang** [comment]
To figure out why users at the same location perceive different SINR, I
suggest you to go to the MmWave3gppPropagationLossModel and MmWave3gppChannel
classes to log out the propagation loss and beam forming gain.

Menglei

**Comment by mychele** [comment]
Closing due to inactivity

**Comment by erazemp** [comment]
Hi I would like to reopen this issue if possible. I'm having a similar issue. I have a simulation with 3 or 4 devices where they are in different containers but all are about the same distance from the enb. The difference in the generated SINR varies by a lot. 
First I thought it was the noise of the devices affecting each other. 

1.) I have modified the mmwave-tcp-example.cc and now I am trying to include the 3gpp model but I got stuck on an "long term" error -> msg: "LongTerm not initialized for this!", in the file mmwave-3gpp-channel.cc. 
I can not figure out how to initialize longTerm variable. 

2.) Is it possible to modify the mmwave-tcp-example.cc this way so I could log out the path loss and beam forming gain?

Thank you!
